### ISSUE TITLE
Add ability to delete app

### ISSUE THREAD

**Issue by CAFxX** [concourse/cf-resource#41]
A common pattern we have is the following:

1. deploy one or more apps in an integration-testing space
2. run integration tests
3. delete the apps deployed in step 1

Currently there's no way to do step 3 with the cf resource, so we have to resort to scripts. Would be nice if the cf resource was extended to allow this kind of operations

**Comment by archgrove** [comment]
+1 on this as a useful thing. I know that Concourse doesn't really have a concept of resource destruction right now, but perhaps something along the lines of https://github.com/ljfranklin/terraform-resource `action: destroy` might be a stopgap?

**Comment by cappyzawa** [comment]
I think it's a useful feature.
But, will this function not make the behavior intuitive?

Image is like bellow?
```yaml
put: cf
params: 
  action: delete # (or deploy)
  manifest: ...
```

If sometimes this resource deploys cf-app, and sometimes delete, this feature might confuse users...

....

If we implement it, we may be able to do it with [cloudfoundry-community/go-cfclient](https://github.com/cloudfoundry-community/go-cfclient).

https://github.com/cloudfoundry-community/go-cfclient/blob/bdb31282431a30516bb8d28628e95dd47e577df6/apps.go#L375-L384

**Comment by sunshineperi** [comment]
Yup, this is definitely better to have.  And I just found this, https://github.com/nulldriver/cf-cli-resource, 
which allow you to do cf delete.
